core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```diff
@@ -248,26 +352,75 @@ private void registerBrokerInZk(KafkaZkClient zkClient, int id) {
         zkClient.registerBroker(brokerInfo);
     }
 
+    private List<String> generateKraftAlterOpts() {
+        return asList("--bootstrap-server", cluster.bootstrapServers(),
+            "--entity-type", "brokers",
+            "--entity-name", "0",
+            "--alter");
+    }
+
+    private void alterAndVerifyConfig(Admin client, Optional<String> brokerId, Map<String, String>... configs) {
+        alterConfigWithKraft(client, brokerId, configs);
+        verifyConfig(client, brokerId, configs);
+    }
+
+    private void alterConfigWithKraft(Admin client, Optional<String> brokerId, Map<String, String>... configs) {
+        String configStr = transferConfigMapsToString(configs);
+        ConfigCommand.ConfigCommandOptions addOpts =
+                new ConfigCommand.ConfigCommandOptions(toArray(alterOpts, entityOp(brokerId), asList("--add-config", configStr)));
+        ConfigCommand.alterConfig(client, addOpts);
+    }
+
+    private void verifyConfig(Admin client, Optional<String> brokerId, Map<String, String>[] configs) throws RuntimeException {
+        ConfigResource configResource = new ConfigResource(ConfigResource.Type.BROKER, brokerId.orElse(defaultBrokerId));
+        client.describeConfigs(singletonList(configResource))
+                .all()
+                .whenComplete((c, e) -> {
+                    if (e != null) {
+                        throw new RuntimeException(e);
+                    }
+                    for (Map<String, String> config : configs) {
+                        assertEquals(config, c.get(configResource).entries());
+                    }
+                });
+    }
+
+    private void deleteAndVerifyConfig(Admin client, Optional<String> brokerId, Set<String> singleton) {
+        ConfigCommand.ConfigCommandOptions deleteOpts =
+                new ConfigCommand.ConfigCommandOptions(toArray(alterOpts, entityOp(brokerId),
+                        asList("--delete-config", String.join(",", singleton))));
+        ConfigCommand.alterConfig(client, deleteOpts);
+        verifyConfig(client, brokerId, new Map[]{Collections.emptyMap()});
+    }
+
     @SafeVarargs
-    static <T> Seq<T> seq(T...seq) {
-        return seq(Arrays.asList(seq));
+    static <T> Seq<T> seq(T... seq) {
```

Comment:
It seems we need this function to pass single `endpoint` to create  `Broker`. Maybe we can add a new constructor to `Broker` to accept single `endpoint`. WDYT?